include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -709,20 +710,29 @@ namespace internal
         for (unsigned int i = 0; i < ranks.size(); ++i)
           {
             MPI_Status status;
-            MPI_Probe(MPI_ANY_SOURCE, 11, communicator, &status);
+            MPI_Probe(
+              MPI_ANY_SOURCE,
+              Utilities::MPI::internal::Tags::fine_dof_handler_view_reinit,
+              communicator,
+              &status);
+
+            std::vector<types::global_dof_index> buffer;
 
             int message_length;
-            MPI_Get_count(&status, MPI_UNSIGNED, &message_length);
+            MPI_Get_count(&status,
+                          Utilities::MPI::internal::mpi_type_id(buffer.data()),
+                          &message_length);
 
-            std::vector<types::global_dof_index> buffer(message_length);
+            buffer.resize(message_length);
 
-            MPI_Recv(buffer.data(),
-                     buffer.size(),
-                     Utilities::MPI::internal::mpi_type_id(buffer.data()),
-                     status.MPI_SOURCE,
-                     11,
-                     communicator,
-                     MPI_STATUS_IGNORE);
+            MPI_Recv(
+              buffer.data(),
+              buffer.size(),
+              Utilities::MPI::internal::mpi_type_id(buffer.data()),
+              status.MPI_SOURCE,
+              Utilities::MPI::internal::Tags::fine_dof_handler_view_reinit,
+              communicator,
+              MPI_STATUS_IGNORE);
```

Comment:
Can you please add a check for the return value `ierr` of `MPI_Recv` (and possibly also in `MPI_Get_count` above), which might have revealed the mismatch in types for 64 bit integers?